contracts/modules/capital/MCR.sol

```diff
@@ -199,53 +140,43 @@ contract MCR is Iupgradable {
 
   }
 
-  /**
-   * @dev Adds MCR Data. Checks if MCR is within valid
-   * thresholds in order to rule out any incorrect calculations
-   */
-  function _addMCRData(
-    uint len,
-    uint64 newMCRDate,
-    bytes4[] memory curr,
-    uint mcrE,
-    uint mcrP,
-    uint vF,
-    uint[] memory _threeDayAvg
-  )
-  internal
-  {
-    uint lowerThreshold = 0;
-    uint upperThreshold = 0;
-
-    if (len > 1) {
-      uint vtp = pool.getPoolValueInEth();
-      (lowerThreshold, upperThreshold) = getThresholdValues(vtp, vF, getAllSumAssurance(), pd.minCap());
-    }
+  function updateMCR(uint poolValueInEth) external onlyInternal {
 
-    if (mcrP > dynamicMincapThresholdx100) {
-      variableMincap = (variableMincap.mul(dynamicMincapIncrementx100.add(10000)).add(minCapFactor.mul(pd.minCap().mul(dynamicMincapIncrementx100)))).div(10000);
+    if (lastUpdateTime + minUpdateTime > now) {
```

Comment:
Initially the buyNXM function used around 95k gas.

if it returns early the function uses 103091
if it does the update it uses 159097